tools/wptrunner/wptrunner/wptlogging.py

```diff
@@ -65,3 +66,41 @@ def __call__(self, data):
             not self.has_log and
             log_levels[data["level"]] <= self.min_level):
             self.has_log = True
+
+
+class QueueHandler(logging.Handler):
+    def __init__(self, queue, level=logging.NOTSET):
+        self.queue = queue
+        logging.Handler.__init__(self, level=level)
+
+    def createLock(self):
+        # The queue provides its own locking
+        self.lock = None
```

Comment:
If it’s not set to None, will something else set it? Would an assert suffice?

Reply:
See https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L2211 `createLock` actually has to create the lock, but the type is effectively `Nullable[Lock]` so this is correct.